src/plot_api/plot_schema.js

```diff
@@ -9,185 +9,287 @@
 
 'use strict';
 
-var Plotly = require('../plotly');
 var Registry = require('../registry');
-var Plots = require('../plots/plots');
 var Lib = require('../lib');
 
-// FIXME polar attribute are not part of Plotly yet
+var baseAttributes = require('../plots/attributes');
+var baseLayoutAttributes = require('../plots/layout_attributes');
+var frameAttributes = require('../plots/frame_attributes');
+var animationAttributes = require('../plots/animation_attributes');
+
+// polar attributes are not part of the Registry yet
 var polarAreaAttrs = require('../plots/polar/area_attributes');
 var polarAxisAttrs = require('../plots/polar/axis_attributes');
 
 var extendFlat = Lib.extendFlat;
 var extendDeep = Lib.extendDeep;
-var extendDeepAll = Lib.extendDeepAll;
 
-var NESTED_MODULE = '_nestedModules',
-    COMPOSED_MODULE = '_composedModules';
+var IS_SUBPLOT_OBJ = '_isSubplotObj';
+var IS_LINKED_TO_ARRAY = '_isLinkedToArray';
+var DEPRECATED = '_deprecated';
+var UNDERSCORE_ATTRS = [IS_SUBPLOT_OBJ, IS_LINKED_TO_ARRAY, DEPRECATED];
```

Comment:
Is it possible to document what these do, perhaps here? A general frustration regarding attribute definitions is that I make up things like `role` or underscored attributes based on copying from similar examples, but I don't really have any idea what they mean or do. In particular:

- `_deprecated` seems like mainly just an annotation. Do the docs use this in deciding how to display things?
- `_isLinkedToArray` seems to have something to do with pluralization and containers, but I haven't figured out what it means beyond that.
- `_isSubplotObj`: Seems like an annotation, perhaps used elsewhere? I can't find that it actually does anything, at least within plotly.js.
- `role`: Not underscored, but in a similar boat. `data_array` seems to be important for extracting data out of the json. Do the other options like `info_array` have any effect? I usually write it because it seems friendly to include, but I don't know if it means anything.

There's probably very little be gained in documenting the internals, but attribute definitions seem like a common thing that people without deep knowledge of the internals probably end up touching pretty quickly.